CONTRIBUTING.md

```diff
@@ -0,0 +1,204 @@
+scripts for running these tests are located in the [s2nTests](s2nTests)
+directory.
+
+# Repository organization and practices
+
+The top-level repository directories are:
+
+* `src` - Source code for the SAWScript interpreter library.
+
+* `saw` - Source for the `saw` executable interpreter for SAWScript.
+
+* `doc` - A tutorial and manual for SAWScript.
+
+* `examples` - Various examples of using SAWScript.
+
+* `intTests` - The central test suite for SAWScript.
+
+* `s2nTests` - Additional tests for SAWScript, confirming that
+  verifications involving the s2n library continue to work.
+
+* `saw-remote-api` - Source for a server that provides a remote API to
+  SAW based on JSON-RPC, and a Python client for that API.
+
+* `saw-core` - Source code for the SAWCore intermediate language used
+  within SAW.
+
+* `saw-core-aig` - A library for translating SAWCore to And-Inverter
+  Graphs.
+
+* `saw-core-coq` - A library for translating SAWCore to Coq's Gallina
+  language.
+
+* `saw-core-sbv` - A library for translating SAWCore predicates to SMT
+  queries using the [SBV](http://leventerkok.github.io/sbv/) library.
+
+* `saw-core-what4` - A library for translating SAWCore predicates to
+  solver queries using the [What4](https://github.com/galoisinc/what4)
+  library.
+
+* `cryptol-saw-core` - A library for translating
+  [Cryptol](https://cryptol.net) code into SAWCore.
+
+* `rme` - A library implementing a Reed-Muller Expansion datatype for
+  representing Boolean formulas.
+
+* `deps` - A location for other dependencies included as Git submodules.
+
+* `crux-mir-comp` - A version of [Crux](https://crux.galois.com/) that
+  can perform modular analysis of Rust code using a form of contracts.
+
+## Branching and merging
+
+Within the `GaloisInc/saw-script` repository, we use a variation on the
+[git-flow
+model](http://nvie.com/posts/a-successful-git-branching-model/) for
+branches and merging with they key difference that our `master` (rather
+than `develop`) branch serves as the cutting edge development branch,
+and our `release-*` (rather than `master`) branches are where only
+pristine, tagged releases are committed.
+
+In short:
+
+- All changes should be developed on branches and then merged into
+  `master` when completed.
+
+- When we reach a feature freeze for a release, we create a new branch
+  prefixed with `release-`, for example `release-0.3`. When the release
+  is made, we merge those changes back into `master` and tag the `HEAD`
+  of the `release` branch.
+
+- If a critical bug emerges in already-released software, we create a
+  branch off of the relevant `release` branch. When the hotfix is
+  complete, we merge those changes back into `master` and add a tag on
+  the `release` branch.
+
+- Merging PRs requires a review from at least one other committer, and
+  requires successful completion of a CI workflow including a wide
+  variety of tests. Branches must be up-to-date with `master` before
+  merging.
```

Comment:
Suggested addition:

- A PR is normally merged by the author of the PR following a Review acceptance and valid CI completion.  This allows the author to determine when they are fully ready for the merge to be performed.  We suggest the use of Mergify (see below) although the merge can also be performed manually, and if you as an author do not have merge permissions, please indicate this in a comment when asking for reviews.

Reply:
Yeah, I like that. Added.